app/app.py

```diff
@@ -1,12 +1,30 @@
-from flask import Flask
+from flask import Flask, jsonify, request
+from os import path
+import subprocess
+import json
+import os
 
 app = Flask(__name__)
 
+# Reading directory path from config.json file
+with open('config.json') as json_data_file:
+    config = json.load(json_data_file)
+    WORKSPACE_DIR = config["path1"]
 
-@app.route('/hello')
-def index():
-    return "Hello, World!"
+# Route for creating workspaces
+@app.route('/workspace/create', methods=['POST'])
+def linchpin_init():
+    data = request.json     # Get request body
+    name = data["name"]
+    cmd = "linchpin init" + " " + name
+    os.chdir(WORKSPACE_DIR)  # Change directory to the one set in config.json
+    if path.isdir(name):
+        return jsonify(status="Workspace already exists")   # Checking if workspace already exists
 
+    else:
+        subprocess.run(cmd, shell=True)  # run the command linchpin init on command line
```

Comment:
If the subprocess command fails to run it would abort the server. 
could you capture the exit status of the command and return it back in response. 
Further, please try to add exception handling to the following line. 

I would like you to capture the exit status and output of the command ran like as follows:
https://stackoverflow.com/questions/2502833/store-output-of-subprocess-popen-call-in-a-string